spec/ddtrace/transport/http/builder_spec.rb

```diff
@@ -13,11 +13,45 @@
   end
 
   describe '#adapter' do
+    context 'given AgentSettings' do
+      subject(:adapter) { builder.adapter(config) }
+
+      let(:config) do
+        instance_double(Datadog::Configuration::AgentSettingsResolver::AgentSettings, adapter: config_adapter)
+      end
+      let(:config_adapter) { :adapter_foo }
+
+      before do
+        allow(Datadog::Configuration::AgentSettingsResolver::AgentSettings).to receive(:===).with(config).and_return(true)
+      end
```

Comment:
Minor: Since this object is rather simple, I wonder if we should just create the actual object (even if half of the fields are `nil`) instead of using a double.

Reply:
Done.